src/lib.rs

```diff
@@ -239,21 +239,66 @@ impl<T> Status<T> {
 }
 
 /// Parser configuration.
-#[derive(Clone, Debug, Default)]
+#[derive(Clone)]
 pub struct ParserConfig {
     allow_spaces_after_header_name_in_responses: bool,
     allow_obsolete_multiline_headers_in_responses: bool,
     allow_multiple_spaces_in_request_line_delimiters: bool,
     allow_multiple_spaces_in_response_status_delimiters: bool,
     ignore_invalid_headers_in_responses: bool,
+
+    version_parser: fn(&mut Bytes) -> Result<u8>,
+}
+
+impl fmt::Debug for ParserConfig {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("ParserConfig")
+         .field("allow_spaces_after_header_name_in_responses", &self.allow_spaces_after_header_name_in_responses)
+         .field("allow_obsolete_multiline_headers_in_responses", &self.allow_obsolete_multiline_headers_in_responses)
+         .field("allow_multiple_spaces_in_request_line_delimiters", &self.allow_multiple_spaces_in_request_line_delimiters)
+         .field("allow_multiple_spaces_in_response_status_delimiters", &self.ignore_invalid_headers_in_responses)
+         .field("ignore_invalid_headers_in_responses", &self.ignore_invalid_headers_in_responses)
+         .finish()
+    }
+}
+
+impl Default for ParserConfig {
+    fn default() -> Self {
+        Self::const_default()
+    }
 }
 
 impl ParserConfig {
+    /// Returns default configuration. Same as `Default::default()` but can be used in const
+    /// context.
+    pub const fn const_default() -> Self {
+        Self {
+            allow_spaces_after_header_name_in_responses: false,
+            allow_obsolete_multiline_headers_in_responses: false,
+            allow_multiple_spaces_in_request_line_delimiters: false,
+            allow_multiple_spaces_in_response_status_delimiters: false,
+            ignore_invalid_headers_in_responses: false,
+            version_parser: HTTP_VERSION_PARSER,
```

Comment:
For the same reason, this also errors, but putting the function pointer in `const` first makes it go away. Interesting.